pkg/scraper/client/resource/client.go

```diff
@@ -60,42 +71,12 @@ func NewClient(config client.KubeletClientConfig) (*kubeletClient, error) {
 	}, nil
 }
 
-type kubeletClient struct {
-	defaultPort       int
-	useNodeStatusPort bool
-	client            *http.Client
-	scheme            string
-	addrResolver      utils.NodeAddressResolver
-	buffers           sync.Pool
-}
-
 var _ client.KubeletMetricsInterface = (*kubeletClient)(nil)
 
-func (kc *kubeletClient) makeRequestAndGetValue(client *http.Client, req *http.Request, value easyjson.Unmarshaler) error {
-	// TODO(directxman12): support validating certs by hostname
-	response, err := client.Do(req)
-	if err != nil {
-		return err
-	}
-	defer response.Body.Close()
-	b := kc.getBuffer()
```

Comment:
Why drop using buffer to avoid copying memory?

Reply:
How to avoid using memory:
```
response, err := client.Do(req)
defer response.Body.Close()
b := kc.getBuffer()
defer kc.returnBuffer(b)
err := io.Copy(b, response.Body)
dec := expfmt.NewDecoder(b, expfmt.FmtText)
...
```
This way we store response body in buffer, and also pass it to decoder.